pkg/database/metrics.go

```diff
@@ -0,0 +1,33 @@
+package database
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"time"
+
+	"github.com/Azure/ARO-RP/pkg/util/recover"
+)
+
+func (db Database) emitMetrics(ctx context.Context) {
+	t := time.NewTicker(time.Minute)
+	defer t.Stop()
+
+	for {
+		i, err := db.OpenShiftClusters.QueueLength(ctx)
+		if err != nil {
+			db.log.Error(err)
+		}
+		defer recover.Panic(db.log)
+
+		db.m.EmitGauge("database.queue.openshiftclusters.length", int64(i), nil)
```

Comment:
don't emit if there was an error before.  move the QueueLength() and EmitGauge() calls into a separate function with normal `if err != nil { return err }` handling

Reply:
I dont think this makes a lot of sense...